client/sse.go

```diff
@@ -53,6 +53,12 @@ func WithSSEReadTimeout(timeout time.Duration) ClientOption {
 	}
 }
 
+func WithAuthToken(token string) ClientOption {
+	return func(sc *SSEMCPClient) {
+		sc.headers["Authorization"] = "Bearer " + token
```

Comment:
If the token expires eventually, then it will need to be updated and this should be thread-safe